Source/Lib/Codec/EbResourceCoordinationProcess.c

```diff
@@ -627,14 +627,16 @@ void* ResourceCoordinationKernel(void *inputPtr)
 
         pictureControlSetPtr->paReferencePictureWrapperPtr = referencePictureWrapperPtr;
 
-        // Give the new Reference a nominal liveCount of 1
+        // Note: the PPCS and its PA reference picture will be released in both EncDec and RateControl kernels.
+        // Give the new Reference a nominal liveCount of 2, meanwhile increase liveCount of PPCS with 1 as it's
+        // already 1 after dequeuing from the PPCS FIFO.
         EbObjectIncLiveCount(
-        	pictureControlSetPtr->paReferencePictureWrapperPtr,
-            2);
+                pictureControlSetPtr->paReferencePictureWrapperPtr,
+                2);
 
         EbObjectIncLiveCount(
-            pictureControlSetWrapperPtr,
-            2);
+                pictureControlSetWrapperPtr,
+                1);
```

Comment:
So with the PR applied, the live count for the PPCS and PA Ref objects in each kernel stage are (numbers in brackets means the current picture is referenced by other picture(s)):

     Obj           |  RC          |  PD           |  EncDec         |   RateControl
     PPCS:         |  2           |  (+1)         |  1 (- 1)        |   0
     PA Ref:       |  2           |  (+1)         |  1 (- 1)        |   0


Without the PR:

     Obj           |  RC          |  IRC         |  PD                  |   RateControl
     PPCS:         |  3           |  2           |  1 (+ 1 - 1)         |   0
     PA Ref:       |  2           |  1           |  0 (+ 1 - 1)         |